File: course_101/exercises/easy_5/ascii_string_value.rb
# Write a method that determines and returns the ASCII string value of a string that is passed in as an argument.
# The ASCII string value is the sum of the ASCII values of every character in the string.
# (You may use String#ord to determine the ASCII value of a character.)

# solution 1
# def ascii_value(str)
#   str.chars.map { |char| char.ord}.sum
# end

# solution 2
def ascii_value(str)
  str.chars.inject(0) { |sum, char| sum + char.ord}
end


puts ascii_value('Four score') == 984
puts ascii_value('Launch School') == 1251
puts ascii_value('a') == 97
puts ascii_value('') == 0

File: course_js210_computational_thinking_and_javascript_programming/exercises/js210_small_problems/easy_1/ascii_string_value.js
// Write a function that determines and returns the ASCII string value of a string passed in as an argument.
// The ASCII string value is the sum of the ASCII values of every character in the string.
// (You may use String.prototype.charCodeAt() to determine the ASCII value of a character.)

function asciiValue(string) {
  return string.split('').map(char => char.charCodeAt(0)).reduce((total, value) => total + value);
  
}

console.time('someFunction');

console.log(asciiValue('Four score'));         // 984

console.timeEnd('someFunction');

// console.log(asciiValue('Launch School'));      // 1251
// console.log(asciiValue('a'));                  // 97
// console.log(asciiValue(''));                   // 0
